internal/conventionalcommits/conventional_commits.go

```diff
@@ -190,17 +190,25 @@ func parseSimpleCommit(commitPart commitPart, commit *gitrepo.Commit, libraryID
 	processFooters(footers)
 
 	var commits []*ConventionalCommit
-	// If the body lines have multiple headers, separate them into  different conventional
-	// commit, all associated with the same commit sha.
-	// Note that we assume single line headers, the 2nd line of headers, if exists,
-	// will be discarded.
+	// Hold the subjects of each commit.
+	var subjects [][]string
+	// If the body lines have multiple headers, separate them into different conventional commit, all associated with
+	// the same commit sha.
 	for _, bodyLine := range bodyLines {
 		header, ok := parseHeader(bodyLine)
 		if !ok {
 			slog.Warn("bodyLine is not a header", "bodyLine", bodyLine, "hash", commit.Hash.String())
+			if len(commits) == 0 {
```

Comment:
I feel it's natural to check the length of subjects because the next sentence is to access `subjects[len(subjects)-1]`.

Reply:
The the length of commits and subjects are always the same.